### PR TITLE
Add support for variadic functions

### PR DESCRIPTION
**Pull request by kant2002**

### CODE REVIEW

File: Cesium.CodeGen/Contexts/FunctionScope.cs [hunk 1]
```diff
@@ -62,7 +62,17 @@ public VariableDefinition ResolveVariable(string identifier)
 
         return variableDefinition;
     }
-    public ParameterInfo? GetParameterInfo(string name) => FunctionInfo.Parameters?.Parameters.FirstOrDefault(p => p.Name == name);
+    public ParameterInfo? GetParameterInfo(string name)
+    {
+        var parametersInfo = FunctionInfo.Parameters;
+        if (parametersInfo is null) return null;
+        if (name == "varargs" && parametersInfo.IsVarArg)
```
Comment (ForNeVeR) [inline Cesium.CodeGen/Contexts/FunctionScope.cs:69]: Let's call it `__varargs` to not clash with any user-defined names.

File: Cesium.CodeGen/Ir/Types/PrimitiveType.cs [hunk 2]
```diff
@@ -149,6 +149,7 @@ Kind switch
                 TargetArchitectureSet.Bit64 => 8,
                 _ => throw new AssertException($"Architecture set not supported: {arch}.")
             },
+            PrimitiveTypeKind.Void => 1,
```
Comment (ForNeVeR) [inline Cesium.CodeGen/Ir/Types/PrimitiveType.cs:152]: Welp, this one is weird. Technically we shouldn't define `sizeof(void)`. Why is that required?

File: Cesium.CodeGen.Tests/CodeGenMethodTests.cs [hunk 3]
```diff
@@ -361,10 +361,11 @@ int main()
 }");
 
     [Fact]
-    public void InvalidPointerWithIntSubtractionTest() => DoesNotCompile(@"int main() {
+    public void InvalidPointerWithIntSubtractionTest() => DoTest(@"int main() {
```
Comment (ForNeVeR) [inline Cesium.CodeGen.Tests/CodeGenMethodTests.cs:364]: _(Minor)_ The name of this test is now misleading: it is no longer `Invalid`, right?

**Review by ForNeVeR — CHANGES_REQUESTED** [review]

File: Cesium.CodeGen/Ir/Types/PrimitiveType.cs (hunk 2 above)
Comment (kant2002) [reply Cesium.CodeGen/Ir/Types/PrimitiveType.cs:152]:
```
void* ptr;
ptr += 8;
```

Comment (ForNeVeR) [reply Cesium.CodeGen/Ir/Types/PrimitiveType.cs:152]: Oh wow, GCC also thinks `sizeof(void)` is 1. Okay.

File: Cesium.Compiler/stdlib/stdarg.h [hunk 4]
```diff
@@ -0,0 +1,9 @@
+#pragma once
+
+typedef void* va_list;
+
+#define va_start(list,...) list = __varargs
+#define va_copy(dest, src) dest = src;
```
Comment (ForNeVeR) [inline Cesium.Compiler/stdlib/stdarg.h:6]: I've decided to remove `;` from here, to force the user to put `;` on their own.

**Review by ForNeVeR — APPROVED** [review]
Thanks!

**Comment by ForNeVeR** [comment]
Whoops, it looks like we have a conflict with your own changes from #481 😅 

Could you please take a look?

**Review by ForNeVeR — APPROVED** [review]
Thanks!
